### ISSUE TITLE
Investigate flakey build

### ISSUE THREAD

**Issue by hcoles** [hcoles/pitest#193]
The build has recently become unstable - possibly as a result of the build sometimes exceeding a time limit on travis.

Maybe the maven verification part of the build could be split out to an alternate ci server such as drone?

**Comment by jasonmfehr** [comment]
The failures such as [168.2](https://travis-ci.org/hcoles/pitest/jobs/60274448) where travis ci kills the build because it has not received any output for 10 minutes I suspect are happening because of the six integration tests I added for the reporting mojo.  I am going to be looking at refactoring some of them to run much quicker.  Until I can send a PR, there are a couple workarounds available.  First, use the travis_wait command.  I tried this command once but could not get it to work and ran out of time to figure it out.  Second, comment out the shouldCopyLatestTimestampedOrNonTimestampedReport and shouldCopyLatestTimestampedReport tests in PitMojoIT.  These two tests run the longest and are the tests I am targeting for refactoring.

**Comment by hcoles** [comment]
@jasonmfehr Thanks - if I encounter any more timeouts I'll add an ignore to one of the timestampedreport tests while I wait on a fix.

**Comment by lkwg82** [comment]
The flaky build is a pain! Can we reduce the tests?

@jasonmfehr I've done some cleanup to PitMojoIT, see #200 especially 2d7986f970c73668b7b3db4af914ef2f034e8c8c

**Comment by lkwg82** [comment]
I have an idea to reduce the wait time in the tests building a site:

Let introduce a custom timestamp format to have seconds instead of minutes as the lowest unit. Then we can wait a second and not wait for the next minute.

What about that?

**Comment by hcoles** [comment]
I've just taken a quick look at these tests - as they stand I don't think they add enough value to justify the pain they cause.

One thought - does the mutationCoverage goal actually need to be run to test the site plugins behaviour? Could the test not just write some date stamped dirs + files to the target directory and then check how the site plugin processes them?

**Comment by lkwg82** [comment]
Removing not needed code is my favourite choice ;)

**Comment by jasonmfehr** [comment]
My original thought process was to run the mutationCoverage goal to guard against changes that would impact the reporting mojo.  In hindsight, I too think those tests are overkill as you all do a great job of maintaining backwards compatibility.  I will leave a couple of the quicker tests that run the mutationCoverage goal and change the others to do their own setup that matches how the pitest maven plugin would write out the pit reports.

I agree that changing the tests is a much better solution in this case rather than adding a custom timestamp format.  I will get them modified and send a new PR.  It's going to take me a week or two though before I can get the updated.

**Comment by jasonmfehr** [comment]
P.S. Thanks @lkwg82 for the heads up on the change!

**Comment by jasonmfehr** [comment]
Pull request #208 has been accepted, and I have not seen any more build failures because of timeouts.  This issue may be closable unless anyone would like more builds to be executed just to be sure.

**Comment by lkwg82** [comment]
I dont oversee the builds ... 

@hcoles what do you think, is this completed so far?

**Comment by hcoles** [comment]
@lkwg82 I still see occasional build failures, but I don't think they relate to @jasonmfehr earlier prs.

I'll close this for now and create a new issue next time a build fails.
